packages/flame/lib/src/components/router_component.dart

```diff
@@ -188,6 +254,18 @@ class RouterComponent extends Component {
     pop();
   }
 
+  /// Local method to bypass [pop]'s assert
+  void _popWithoutMinimumAssert() {
+    assert(
+      _routeStack.isNotEmpty,
+      'There must be a route to pop.',
+    );
```

Comment:
I think this assert can be removed. Because
- it can never trigger as far as we know (there's always at least one route on the stack);
- even if the stack becomes empty through some unknown means, the error will trigger on the line below anyways;
- the method's name says "WithoutAssert", yet here's the assert, so it's a bit awkward.

Reply:
Yeah, I was kind of torn on this one. I didn't have it at first, hence the name but I was concerned it would somehow get called on an empty stack and I didn't want to throw an error if it could be asserted. I agree it should never be empty, but didn't know if it should be overly cautious or not. I can remove it.